config/sampler_config.go

```diff
@@ -221,6 +289,36 @@ type RulesBasedDownstreamSampler struct {
 	DeterministicSampler      *DeterministicSamplerConfig      `json:"deterministicsampler" yaml:"DeterministicSampler,omitempty"`
 }
 
+func (r *RulesBasedDownstreamSampler) GetSamplingFields() []string {
+	fields := make(generics.Set[string], 0)
+
+	if r.DeterministicSampler != nil {
+		fields.Add(r.DeterministicSampler.GetSamplingFields()...)
+	}
+
+	if r.DynamicSampler != nil {
+		fields.Add(r.DynamicSampler.GetSamplingFields()...)
+	}
+
+	if r.EMADynamicSampler != nil {
+		fields.Add(r.EMADynamicSampler.GetSamplingFields()...)
+	}
+
+	if r.EMAThroughputSampler != nil {
+		fields.Add(r.EMAThroughputSampler.GetSamplingFields()...)
+	}
+
+	if r.WindowedThroughputSampler != nil {
+		fields.Add(r.WindowedThroughputSampler.GetSamplingFields()...)
+	}
+
+	if r.TotalThroughputSampler != nil {
+		fields.Add(r.TotalThroughputSampler.GetSamplingFields()...)
+	}
+
+	return fields.Members()
```

Comment:
I think only one sampler implementation can be set, so we could return the result of `GetSamplingFields` instead of adding to an intermediate variable then returning that.

```suggestion
	if r.DeterministicSampler != nil {
		return r.DeterministicSampler.GetSamplingFields()
	}

	if r.DynamicSampler != nil {
		return r.DynamicSampler.GetSamplingFields()
	}

	if r.EMADynamicSampler != nil {
		return r.EMADynamicSampler.GetSamplingFields()
	}

	if r.EMAThroughputSampler != nil {
		return r.EMAThroughputSampler.GetSamplingFields()
	}

	if r.WindowedThroughputSampler != nil {
		return r.WindowedThroughputSampler.GetSamplingFields()
	}

	if r.TotalThroughputSampler != nil {
		return r.TotalThroughputSampler.GetSamplingFields()
	}

	return make([]string, 0)
```